clone: generate unrolled code for cloning messages

misberner opened the pull request:
This adds a new `clone` option to the generator that, when enabled, generates the following optimized methods for each message type:
- `func (p *YourProto) CloneVT() *YourProto`, which returns a cloned message, i.e., the resulting message is equal to `p` but does not share any memory.
- `func (p *YourProto) CloneGenericVT() proto.Message`, which does the same as the above, with the benefit of offering a type-independent signature such that it can be invoked via an interface type assertion on any `proto.Message` (in a dynamic setting where the type is not known at compile time, `CloneVT` can only be accessed via reflection/`MethodByName`)

Fixes #45. (actually, just one half, I can easily add the `CopyVT` logic if this is desired)

vmg commented:
Hey @misberner! Thanks for the contribution. The code looks great! I'm a bit concerned about the clone helpers which are dynamically loaded on `init`. If you compare it to the other implementations in `vtprotobuf`, you can see that we're performing an `interface` check inline to see if a fast method is available, and that has _never_ shown up in profiles.

I would prefer if the copy code also did the same instead of pre-loading the helpers in an `init`... But if you can show me in a benchmark that it's gonna have a significant effect in performance, I'll change my mind!

misberner commented:
Thanks for the feedback, @vmg ! I have just removed the helper function logic. I introduced them because performance was initially worse than I had hoped, but that turned out to be due to something else entirely. I then left them in because I thought they wouldn't hurt, but you are right that it's better to keep it consistent with the other generators.

vmg reviewed (approved):
Brilliant. Thank you so much!
